wallet/createtx.go

```diff
@@ -1228,6 +1233,568 @@ func (w *Wallet) purchaseTickets(req purchaseTicketRequest) ([]*chainhash.Hash,
+		totalAdded += credit.Amount
+		forSigning = append(forSigning, credit)
+	}
+
+	if totalAdded < ticketPrice {
+		return nil, ErrSStxNotEnoughFunds
+	}
+
+	// This is the total change that should be paid
+	totalChange := totalAdded - ticketPrice - feeEst
+
+	// total commitment
+	totalCommitment := ticketPrice + feeEst
+
+	for _, credit := range eligible {
+		prop := float64(credit.Amount) / float64(totalAdded)
+
+		// for each eligible credit, change and commitment should be in proportion to credit size
+
+		changeWithThisCredit := totalChange.MulF64(prop)
+		commitmentWithThisCredit := totalCommitment.MulF64(prop)
+
+		pkScript, err := txscript.GenerateSStxAddrPush(subsidyAddress, commitmentWithThisCredit, uint16(defaultTicketFeeLimits))
+		if err != nil {
+			return nil, fmt.Errorf("cannot create txout script: %s", err)
+		}
+		txout := wire.NewTxOut(int64(0), pkScript)
+		msgtx.AddTxOut(txout)
+
+		// fetch change address
+		changeAddr, err := w.newChangeAddress(w.persistReturnedChild(dbtx), udb.DefaultAccountNum)
+		if err != nil {
+			return nil, err
+		}
+
+		// Add the changeSStx Output. This randomizes the index of output
+		pkScript, err = txscript.PayToSStxChange(changeAddr)
+		if err != nil {
+			return nil, fmt.Errorf("cannot create txout script: %s", err)
+		}
+
+		msgtx.AddTxOut(wire.NewTxOut(int64(changeWithThisCredit), pkScript))
+	}
+	return forSigning, nil
+}
+
+func (w *Wallet) estimateFee(eligible []udb.Credit, votingAddress dcrutil.Address,
+	ticketPrice dcrutil.Amount) (dcrutil.Amount, error) {
+
+	msgtx := wire.NewMsgTx()
+
+	scriptSizers := []txsizes.ScriptSizer{}
+
+	for range eligible {
+		scriptSizers = append(scriptSizers, txsizes.P2SHScriptSize)
+	}
+
+	pkScript, err := txscript.PayToSStx(votingAddress)
+	if err != nil {
+		return dcrutil.Amount(0), fmt.Errorf("cannot create txout script: %s", err)
+	}
+	txout := wire.NewTxOut(int64(ticketPrice), pkScript)
+	msgtx.AddTxOut(txout)
+
+	for _, credit := range eligible {
+		pkScript, err := txscript.GenerateSStxAddrPush(votingAddress, credit.Amount, uint16(defaultTicketFeeLimits))
+		if err != nil {
+			return dcrutil.Amount(0), fmt.Errorf("cannot create txout script: %s", err)
+		}
+		txout := wire.NewTxOut(int64(0), pkScript)
+		msgtx.AddTxOut(txout)
+
+		// Add the changeSStx Output. This randomizes the index of output
+		pkScript, err = txscript.PayToSStxChange(votingAddress)
+		if err != nil {
+			return dcrutil.Amount(0), fmt.Errorf("cannot create txout script: %s", err)
+		}
+		msgtx.AddTxOut(wire.NewTxOut(int64(0), pkScript))
+	}
+
+	szEst := txsizes.EstimateSerializeSize(scriptSizers, msgtx.TxOut, true)
+	feeEst := txrules.FeeForSerializeSize(w.RelayFee(), szEst)
+
+	return feeEst, nil
+
+}
+
+func (w *Wallet) forceSplit(req purchaseTicketRequest, numTickets int) (*chainhash.Hash, error) {
+	ticketHashes, err := w.purchaseTicketsSplit(req, numTickets)
+	if err != nil {
+		return nil, err
+	}
+	return ticketHashes[0], nil
```

Comment:
I'm still not seeing this at a high level, but what does it mean when `numTickets > 1`?

Reply:
for now, numTickets can only be 1. the variable is used for future purposes

Reply:
After this was built, it was discovered that though the function works, it is inefficient to use more then one input due to the 1:3 input to output ratio requirement for pool ticket splitting. If the 1:3 ratio requirement is removed in the future, usage with more then one input can be used.